source/vision/visionHandler.py

```diff
@@ -348,3 +356,9 @@ def initialFocus(self) -> None:
 			# focus/review hasn't yet been initialized.
 			return
 		self.handleGainFocus(api.getFocusObject())
+	
+	def initialNavigatorObject(self) -> None:
+		if not api.getDesktopObject():
+			# focus/review hasn't yet been initialized.
+			return
+		self.handleReviewMove(api.getNavigatorObject())

```

Comment:
This doesn't look right to me - `self.handleReviewMove` takes a `Context` not an `NVDAObject`.
I also suggest moving the try/except to this function

Reply:
> This doesn't look right to me - self.handleReviewMove takes a Context not an NVDAObject.

I have fixed it in d7506dcaa7814319f2af6873aec499fe7b2ac262.
It had no impact on the final behavior for NVDAHighlighter provider however since this parameter is not used there; but it may be used in other providers in add-ons or future ones in core of course.

> I also suggest moving the try/except to this function

I have defined `initialNavigatorObject` on the same pattern as `initialFocus`. Then, do you expect both try/except to be moved in the functions?

Reply:
Perhaps not, but I think it should be noted in the `initialFocus` docstring that it may raise an exception from `handleGainFocus` and the same for `initialNavigatorObject` and `handleReviewMove`. You could catch and re-raise the exception as well to make it explicit.

Reply:
Not sure what to add exactly in the docstring of `initialFocus` and `initialNavigatorObject`.

What about:
"This function notifies one or more extension point; thus it can raise any type of exception."

It does not seem to bring more useful information however.

If you can make a suggestion, maybe I will understand better what is missing.
Thanks.

Note that the extension points can be the ones of NVDAHighlighter but also the ones of visual enhancements provided in add-ons.

Reply:
That example sounds fine. It's helpful to know explicitly that the function may throw an exception.

Reply:
Fixed in 9e2a03e07bbc9fcb48dc9184cdbab1e8c2a52bab.